### PR TITLE
Updates payment intent parameters

### PR DESCRIPTION
**Pull request by gokulchandra**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://stripe.com/docs/api/payment_intents/create?lang=node
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by armanio123** [comment]
Can I get an approval from any of the reviewers? This PR introduces a bunch of breaking changes.

**Comment by pushplay** [comment]
It's hard to evaluate because there are semantic changes mixed in with style changes.

**Comment by armanio123** [comment]
@pushplay I feel you. Hiding the whitespace changes helped me taking a quick look. You can find the setting on the gear icon on top.

File: types/stripe/index.d.ts [hunk 1]
```diff
@@ -5148,7 +5154,7 @@ declare namespace Stripe {
             transfer_data?: IPaymentIntentDataTransferDataOptions;
         }
 
-        type PaymentIntentCancellationReason = 'duplicate' | 'fraudulent' | 'requested_by_customer' | 'failed_invoice';
+        type PaymentIntentCancellationReason = 'duplicate' | 'fraudulent' | 'requested_by_customer' | 'abandoned';
```
Comment (pushplay) [inline types/stripe/index.d.ts]: We need separate types for reasons that can be given for cancellation (which are the ones here) and what can be on the object which includes some reasons internal to Stripe.

Per the doc:
> Reason for cancellation of this PaymentIntent, either user-provided (duplicate, fraudulent, requested_by_customer, or abandoned) or generated by Stripe internally (failed_invoice, void_invoice, or automatic).

**Review by pushplay — CHANGES_REQUESTED** [review]

File: types/stripe/index.d.ts (hunk 1 above)
Comment (gokulchandra) [reply types/stripe/index.d.ts]: Updated!

**Review by pushplay — APPROVED** [review]
That looks like a good solution to me.  Thanks.

**Comment by bruun** [comment]
@gokulchandra This broke `IPaymentIntentCreationOptions` by removing `transfer_data` which is accepted at create time: https://stripe.com/docs/api/payment_intents/create

**Comment by pushplay** [comment]
Take a look at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39157

**Pull request merged**